node.cpp

```diff
@@ -0,0 +1,192 @@
+#include <mutex>
+
+#include "node.h"
+#include "roles.h"
+
+std::once_flag init_once, cleanup_once;
+
+void Node::initialize()
+{
+  std::call_once(init_once, []() { xmlInitParser(); });
+}
+
+void Node::cleanup()
+{
+  std::call_once(cleanup_once, []() { xmlCleanupParser(); });
```

Comment:
in theory you should never call cleanup, unless you don't want to use the library anymore